src/rust/catnap/queue.rs

```diff
@@ -57,6 +69,28 @@ impl CatnapQueue {
     pub fn get_socket(&self) -> &Socket {
         &self.socket
     }
+
+    /// Adds a new operation to the list of pending operations on this queue.
+    pub fn add_pending_op(&mut self, scheduler_handle: &SchedulerHandle, yielder_handle: &YielderHandle) {
+        self.pending_ops
+            .insert(scheduler_handle.clone(), yielder_handle.clone());
+    }
+
+    /// Removes an operation from the list of pending operations on this queue.
+    pub fn remove_pending_op(&mut self, scheduler_handle: &SchedulerHandle) {
+        self.pending_ops.remove(scheduler_handle);
+    }
+
+    pub fn cancel_pending_ops(&mut self, cause: Fail) {
+        for (mut scheduler_handle, mut yielder_handle) in self.pending_ops.drain() {
+            if !scheduler_handle.has_completed() {
+                trace!("Waking running coroutines and cancelling them");
```

Comment:
Could you prefix log messages with the function name? Otherwise, when running with logs enabled we cannot easily figure out where those messages come from. Also, consider adding the parameters that were passed to the logged function, to help on understanding the context.